contentcuration/contentcuration/frontend/channelEdit/vuex/clipboard/actions.js

```diff
@@ -323,50 +395,57 @@ export function resetSelectionState(context) {
   });
 }
 
-export function deleteClipboardNode(context, { clipboardNodeId, ancestorId = null }) {
-  if (context.getters.isClipboardNode(clipboardNodeId)) {
-    return Clipboard.delete(clipboardNodeId).then(() => {
-      context.commit('REMOVE_CLIPBOARD_NODE', { id: clipboardNodeId });
-    });
+export function deleteClipboardNode(context, { id }) {
+  const ancestor = context.getters.getClipboardAncestorNode(id);
+  if (!ancestor) {
+    throw new Error('Unexpected null ancestor');
   }
-  const ancestor = context.state.clipboardNodesMap[ancestorId];
 
-  const contentNode = context.rootGetters['contentNode/getContentNode'](clipboardNodeId);
+  // if ID references node that is child of clipboard root, then we can
+  // delete it all, likewise with a legacy node
+  if (ancestor.id === id || context.getters.isLegacyNode(id)) {
+    return Clipboard.delete(id).then(() => {
+      context.commit('REMOVE_CLIPBOARD_NODE', { id });
+    });
+  }
 
+  // The node must be a new version clipboard node that's not the top level ancestor, so
+  // update the ancestor's exclusion data
+  const contentNode = context.getters.getContentNodeForRender(id);
   if (contentNode) {
-    const excluded_descendants = {
-      ...get(ancestor, ['extra_fields', 'excluded_descendants'], {}),
-      [contentNode.node_id]: true,
-    };
-    const update = {
-      extra_fields: {
-        ...ancestor.extra_fields,
-        excluded_descendants,
-      },
-    };
+    const { extra_fields } = addExcludedNode(ancestor, contentNode);
     // Otherwise we have a non clipboard node
-    return Clipboard.update(ancestorId, update).then(() => {
-      context.commit('ADD_CLIPBOARD_NODE', {
-        ...ancestor,
-        ...update,
-      });
+    return Clipboard.update(ancestor.id, { extra_fields }).then(() => {
+      context.commit('ADD_CLIPBOARD_NODE', ancestor);
+      context.commit('REMOVE_CLIPBOARD_NODE', { id });
     });
   }
 }
 
-export function deleteClipboardNodes(context, selectionIds) {
+export function deleteClipboardNodes(context, ids) {
   return Promise.all(
-    selectionIds.map(selectionId => {
-      return deleteClipboardNode(context, {
-        clipboardNodeId: selectionId.split('-')[0],
-        ancestorId: selectionId.split('-')[1],
-      });
+    ids.map(id => {
+      // Going through list of ids to delete, we make sure to detect if the parent or top-level
+      // ancestor is also included in the array to avoid unnecessary delete calls
+      const clipboardNode = context.state.clipboardNodesMap[id];
+      const ancestor = context.getters.getClipboardAncestorNode(id);
+      if (
+        (clipboardNode && ids.includes(clipboardNode.parent)) ||
+        (ancestor && ancestor.id !== id && ids.includes(ancestor.id))
+      ) {
+        context.commit('REMOVE_CLIPBOARD_NODE', { id });
+        return Promise.resolve();
+      }
+
+      return deleteClipboardNode(context, { id });
     })
```

Comment:
This fixed issues with deleted nodes reappearing, by ensuring we don't trigger deletes for nodes that are descendants of nodes we're also going to delete.